pkg/validation/spec/swagger.go

```diff
@@ -164,14 +164,14 @@ func (s *SchemaOrBool) UnmarshalJSON(data []byte) error {
 	}
 
 	var nw SchemaOrBool
-	if len(data) >= 4 {
-		if data[0] == '{' {
-			var sch Schema
-			if err := json.Unmarshal(data, &sch); err != nil {
-				return err
-			}
-			nw.Schema = &sch
+	if data[0] == '{' {
+		var sch Schema
+		if err := json.Unmarshal(data, &sch); err != nil {
+			return err
 		}
+		nw.Schema = &sch
+		nw.Allows = true
+	} else {
 		nw.Allows = !(data[0] == 'f' && data[1] == 'a' && data[2] == 'l' && data[3] == 's' && data[4] == 'e')
```

Comment:
huh, wondering how this guarded against a value of `fals` passing the `len(data) >= 4` check and panicking on `data[4] == 'e' `

Reply:
Yeah, I think this whole thing was just very wrong

Reply:
But it's very unlikely that you will have a json that finishes in `fals`

Reply:
fixed